tests/cases/unittests/session.ts

```diff
@@ -0,0 +1,454 @@
+                    responseRequired: false
+                });
+                expect(lastSent).to.deep.equal({
+                    command: CommandNames.Configure,
+                    type: "response",
+                    success: true,
+                    request_seq: 0,
+                    seq: 0,
+                    body: undefined
+                });
+            });
+        });
+
+        describe("onMessage", () => {
+            it("should not throw when commands are executed with invalid arguments", () => {
+                let i = 0;
+                for (name in CommandNames) {
+                    if (!Object.prototype.hasOwnProperty.call(CommandNames, name)) {
+                        continue;
+                    }
+                    const req: protocol.Request = {
+                        command: name,
+                        seq: i++,
+                        type: "command"
+                    };
+                    session.onMessage(JSON.stringify(req));
+                    req.seq = i++;
+                    req.arguments = {};
+                    session.onMessage(JSON.stringify(req));
+                    req.seq = i++;
+                    req.arguments = null;
+                    session.onMessage(JSON.stringify(req));
+                    req.seq = i++;
+                    req.arguments = "";
+                    session.onMessage(JSON.stringify(req));
+                    req.seq = i++;
+                    req.arguments = 0;
+                    session.onMessage(JSON.stringify(req));
+                    req.seq = i++;
+                    req.arguments = [];
+                    session.onMessage(JSON.stringify(req));
+                }
+                session.onMessage("GARBAGE NON_JSON DATA");
+            });
+            it("should output the response for a correctly handled message", () => {
+                const req: protocol.ConfigureRequest = {
+                    command: CommandNames.Configure,
+                    seq: 0,
+                    type: "command",
+                    arguments: {
+                        hostInfo: "unit test",
+                        formatOptions: {
+                            newLineCharacter: "`n"
+                        }
+                    }
+                };
+
+                session.onMessage(JSON.stringify(req));
+
+                expect(lastSent).to.deep.equal(<protocol.ConfigureResponse>{
+                    command: CommandNames.Configure,
+                    type: "response",
+                    success: true,
+                    request_seq: 0,
+                    seq: 0,
+                    body: undefined
+                });
+            });
+        });
+
+        describe("exit", () => {
+            it("is a noop which can be handled by subclasses", () => {
+                session.exit(); // Does nothing, should keep running tests
+                expect(session).to.exist;

```

Comment:
Honestly the line probably shouldn't be there at all... Really, I effectively wanted an `assert(true)` after calling session.exit() to indicate that the process should still be running. I'm not really sure how to test that a function which should be a noop is, in fact, still a noop. (Beyond literally stringifying the function and checking it's source, which seems extremely tautological)

Actually, in every case `Session` gets subclassed - maybe we should just mark Session as `abstract` and `exit()` as an abstract method (then remove the test)? I think that may be more in line with what is supposed to occur.